libs/MobileSync/MobileSync/Classes/Target/SFLayoutSyncDownTarget.m

```diff
@@ -63,56 +72,78 @@ - (instancetype)init {
     return self;
 }
 
-+ (SFLayoutSyncDownTarget *)newSyncTarget:(NSString *)objectType layoutType:(NSString *)layoutType {
++ (SFLayoutSyncDownTarget *)newSyncTarget:(NSString *)objectType
+                               layoutType:(NSString *)layoutType {
+    return [SFLayoutSyncDownTarget newSyncTarget:objectType formFactor:nil layoutType:layoutType mode:nil recordTypeId:nil];
+}
+
++ (SFLayoutSyncDownTarget *)newSyncTarget:(NSString *)objectAPIName
+                               formFactor:(NSString *)formFactor
+                               layoutType:(NSString *)layoutType
+                                     mode:(NSString *)mode
+                             recordTypeId:(NSString *)recordTypeId {
     SFLayoutSyncDownTarget *syncTarget = [[SFLayoutSyncDownTarget alloc] init];
     syncTarget.queryType = SFSyncDownTargetQueryTypeLayout;
-    syncTarget.objectType = objectType;
+    syncTarget.objectAPIName = objectAPIName;
+    syncTarget.formFactor = formFactor;
     syncTarget.layoutType = layoutType;
+    syncTarget.mode = mode;
+    syncTarget.recordTypeId = recordTypeId;
     return syncTarget;
 }
 
 - (NSMutableDictionary *)asDict {
     NSMutableDictionary *dict = [super asDict];
-    dict[kSFSyncTargetObjectType] = self.objectType;
+    dict[kSFSyncTargetObjectType] = self.objectAPIName;
+    dict[kSFSyncTargetFormFactor] = self.formFactor;
```

Comment:
@bhariharan Some of these properties are nullable right? Should guard the assignment to dictionary.